docs/user/user/adding-glean-to-your-project/javascript.md

```diff
@@ -0,0 +1,133 @@
+
+The Glean.js package is distributed as an npm package [`@mozilla/glean`](https://www.npmjs.com/package/@mozilla/glean).
+
+Install Glean.js in your Javascript project, by running:
+
+```bash
+npm install @mozilla/glean
+```
+
+Then import Glean into your project:
+
+```js
+// Glean.js for webextensions
+//
+// esm
+import Glean from "@mozilla/glean/webext";
+// cjs
+const { default: Glean } = require("@mozilla/glean/webext");
+```
+
+### Common import errors
+
+#### "Cannot find module '@mozilla/glean'"
+
+Glean.js does not have a [`main`](https://nodejs.org/api/packages.html#packages_main_entry_point_export) package entry point.
+Instead it relies on a series of entry points depending on the platform you are targeting.
+ 
+In order to import Glean use:
+
+```js
+import Glean from '@mozilla/glean/{your-platform}'
+```
+ 
+#### "Module not found: Error: Can't resolve '@mozilla/glean/webext' in '...'"
+ 
+Glean.js relies on Node.js' [subpath exports](https://nodejs.org/api/packages.html#packages_subpath_exports)
+feature to define multiple package entry points.
+ 
+Please make sure that you are using a supported Node.js runtime and also make sure the tools you are using support this Node.js feature.
+
+## Setting up metrics and pings code generation
+
+In Javascript, the metrics and pings definitions must be parsed at build time.
+The `@mozilla/glean` package exposes [glean_parser](https://github.com/mozilla/glean_parser) through the `glean` script.
+
+To parse your YAML registry files using this script, define a new script in your `package.json` file:
+
+```json
+{
+  // ...
+  "scripts": {
+    // ...
+    "build:glean": "glean translate path/to/metrics.yaml path/to/pings.yaml -f javascript -o path/to/generated",
+    // Or, if you are building for a Typescript project
+    "build:glean": "glean translate path/to/metrics.yaml path/to/pings.yaml -f typescript -o path/to/generated"
+  }
+}
+```
+
+Then run this script by calling:
+
+```bash
+npm run build:glean
+```
+
+## Automation steps
+
+### Documentation
+
+{{#include ../../../shared/blockquote-warning.html}}
+
+##### Prefer using the Glean Dictionary
+
+> While it is still possible to generate Markdown documentation, if working on a public Mozilla project rely on the [Glean Dictionary] for documentation.
+> Your product will be automatically indexed by the Glean Dictionary after it gets enabled in the pipeline.
+
+One of the commands provided by `glean_parser` allows users to generate Markdown documentation based on the contents of their YAML registry files.
+To perform that translation, use the `translate` command with a different output format, as shown below.
+
+In your `package.json`, define the following script:
+
+```json
+{
+  // ...
+  "scripts": {
+    // ...
+    "docs:glean": "glean translate path/to/metrics.yaml path/to/pings.yaml -f markdown -o path/to/docs",
+  }
+}
+```
+
+Then run this script by calling:
+
+```bash
+npm run docs:glean
+```
+
+### YAML registry files linting
```

Comment:
nit: In the summary these pages are called simply "YAML files". I am not saying to change them here, but we should reach consensus on terminology and stick to it. This can be a discussion for a follow up.

Reply:
The [original terminology is YAML registry files](https://docs.google.com/document/d/186GI_2ljdqWqWTBk5P3QjHD3BwpMyd8kRayRAuz1-dk/edit#heading=h.oljvbeczadae). Let me change the summary